addons/mail/static/src/models/dialog/dialog.js

```diff
@@ -12,10 +12,10 @@ function factory(dependencies) {
          * @override
          */
         delete() {
+            super.delete();
```

Comment:
Don't know if this is a necessary change or not. If it's just a preference, I prefer revert because order makes only sense in context with update cycle. Without this, this code makes no sense at all.

Reply:
The line is useless actually, unregister is doing unlink and the relation is causal. The problem is that we don't want an infinite loop, we don't want the "already deleted error", and we must call super for the delete (from causal) to have an effect.

I think the real fix is to not have to do unregister in the first place. manager should be refactored so that unlink on delete properly does equivalent of unregister through computes.